src/modules/Vocabulary/components/List/components/ModalConfirmDownload/selectors.ts

```diff
@@ -0,0 +1,44 @@
+/*
+ *
+ * Copyright 2018 Odysseus Data Services, inc.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ * Company: Odysseus Data Services, Inc.
+ * Product Owner/Architecture: Gregory Klebanov
+ * Authors: Alexandr Saltykov, Pavel Grafkin, Vitaly Koulakov, Anton Gackovka
+ * Created: March 3, 2017
+ *
+ */
+
+import {createSelector} from 'reselect';
+import {get} from 'lodash';
+
+interface Version {
+    value: string;
+    label: string;
+}
+
+const getRawVocabVersions = (state: Object) => get(state, 'vocabulary.versions.queryResult') || [];
+
+const getVocabVersions = createSelector(
+    getRawVocabVersions,
+    (rawResults: Array<Version>) => rawResults.map((version: Version) => ({
```

Comment:
I'm not sure about this, but why do you need to create a copy of results? and then make a copy of each version object? is it necessary?

Reply:
At the moment, the backend provides the JSON with the same fields, but the requirement here is not clear. I suppose we are going to remap the backend response. I can remove it for now.